File: Cpp/Linked List/Delete in a Singly Linked List.cpp
//https://www.geeksforgeeks.org/problems/delete-a-node-in-single-linked-list/1
Node* deleteNode(Node *head,int x)
{
    //Your code here
    if(x==1){
        return head->next;
    }
    Node *temp=head;
    while(temp!=NULL && x>2){
        temp=temp->next;
        x--;
    }
    if(temp->next!=NULL){
        temp->next=temp->next->next;
    }
    else{
        temp->next=NULL;
    }
    return head;
}

File: Python/LinkedList/Delete in a Singly Linked List.py
//https://www.geeksforgeeks.org/problems/delete-a-node-in-single-linked-list/1
def deleteNode(head, x):
    if x == 1:
        return head.next

    temp = head
    for _ in range(x - 2):
        if temp is None:
            return head
        temp = temp.next

    if temp is not None and temp.next is not None:
        temp.next = temp.next.next

    return head
